test/jit/test_peephole.py

```diff
@@ -506,13 +506,18 @@ def foo(x: int, y: int):
         FileCheck().check("aten::add").run(foo.graph)
 
     def test_optimize_out_comparison_same_value(self):
-        @torch.jit.script
         def foo(x: int):
             return x == x, x != x
 
-        self.run_pass("peephole", foo.graph)
-        FileCheck().check_not("aten::eq").check_not("aten::neq").run(foo.graph)
-        self.assertEqual(foo(1), (True, False))
+        def foo2(x: List[int]):
+            return x == x, x != x
+
+        inps = 1, [2, 3]
+        for func, inp in zip([foo, foo2], inps):
```

Comment:
rewriting the zip -> [[foo, 1], [foo2, [2, 3]] honestly would be shorter and clearer.